lib/selinux/selinux_linux.go

```diff
@@ -103,6 +137,20 @@ func CheckConfiguration(ensureEnforced bool, logger *slog.Logger) error {
 		}
 	}
 
+	// ensure we are running under the correct domain.
+	label, err := ocselinux.CurrentLabel()
+	if err != nil {
+		return trace.Wrap(err, "failed to get SELinux context")
+	}
+	seCtx, err := parseLabel(label)
+	if err != nil {
+		return trace.Wrap(err, "failed to parse SELinux context")
+	}
+	if seCtx.domain != domain {
```

Comment:
Can we include some test coverage for this logic?

Reply:
I could mock the logic that fetches the current context, but I honestly don't really think it'd bring a lot of value. I'd mainly just be testing that the context is parsed correctly, which is mainly handled by the selinux dep, and detect when the wrong domain is used. I'm planning on adding test cases for all failure conditions in `CheckConfiguration` in the automated SELinux tests where it will be much more valuable imo.